src/pages/iv-categories/index.js

```diff
@@ -0,0 +1,278 @@
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import CategoryWindow from './window/CategoryWindow'
+import CustomComboBox from 'src/components/Inputs/CustomComboBox'
+
+const Category = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+  const [selectedReport, setSelectedReport] = useState(null)
+  const [reportStore, setReportStore] = useState([])
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    invalidate,
+    filterBy,
+    clearFilter,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: InventoryRepository.Category.page,
+    datasetId: ResourceIds.IvCategories,
+
+    filter: {
+      endpointId: InventoryRepository.Category.snapshot,
+      filterFn: fetchWithSearch
+    }
+  })
+
+  async function fetchWithSearch({ filters, pagination }) {
+    return filters.qry
+      ? await getRequest({
+          extension: InventoryRepository.Category.snapshot,
+          parameters: `_filter=${filters.qry}`
+        })
+      : await fetchGridData(pagination)
+  }
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    const response = await getRequest({
+      extension: InventoryRepository.Category.page,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_name=`
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  const columns = [
+    {
+      field: 'caRef',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'name',
+      headerName: _labels.name,
+      flex: 1
+    },
+    {
+      field: 'parentName',
+      headerName: _labels.parentCat,
+      flex: 1
+    },
+    {
+      field: 'measurementName',
+      headerName: _labels.measurementSchedule,
+      flex: 1
+    },
+    ,
+    {
+      field: 'nraRef',
+      headerName: _labels.nra,
+      flex: 1
+    },
+    {
+      field: 'taxName',
+      headerName: _labels.vatSchedule,
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    openForm()
+  }
+
+  const del = async obj => {
+    try {
+      await postRequest({
+        extension: InventoryRepository.Category.del,
+        record: JSON.stringify(obj)
+      })
+      invalidate()
+      toast.success(platformLabels.Deleted)
+    } catch (error) {}
+  }
+
+  function openForm(recordId) {
+    stack({
+      Component: CategoryWindow,
+      props: {
+        labels: _labels,
+        recordId: recordId,
+        maxAccess: access
+      },
+      width: 800,
+      height: 600,
+      title: _labels.categories
+    })
+  }
+
+  const edit = obj => {
+    openForm(obj?.recordId)
+  }
+
+  useEffect(() => {
```

Comment:
why this useEffect is affected by ResourceIds.IvCategories, can't be added with empty dependency?

Reply:
it should works only if there is resourceId ,